test/e2e/e2e_test.go

```diff
@@ -108,68 +133,119 @@ func TestTaskRun(t *testing.T) {
 		}
 	})
 
-	t.Run("TaskRun Cleanup", func(t *testing.T) {
+	t.Run("check deletion", func(t *testing.T) {
 		if err := wait.PollImmediate(1*time.Second, 1*time.Minute, func() (done bool, err error) {
-			tr, err := c.TaskRuns(ns).Get(ctx, tr.GetName(), metav1.GetOptions{})
-			if k8serrors.IsNotFound(err) {
-				return true, nil
+			_, err = tc.TaskRuns(defaultNamespace).Get(ctx, tr.GetName(), metav1.GetOptions{})
+			if err != nil {
+				if k8serrors.IsNotFound(err) {
+					return true, nil
+				}
+				t.Errorf("Error getting PipelineRun: %v", err)
+				return false, err
 			}
-			t.Logf("Get: %+v, %v", tr.GetName(), err)
 			return false, nil
 		}); err != nil {
-			t.Fatalf("error waiting TaskRun to be deleted: %v", err)
+			t.Fatalf("Error waiting for TaskRun deletion: %v", err)
+		}
+	})
+
+	t.Run("check result", func(t *testing.T) {
+		if resName == "" {
+			t.Skip("Result name not found")
+		}
+		_, err = gc.GetResult(context.Background(), &resultsv1alpha2.GetResultRequest{Name: resName})
+		if err != nil {
+			t.Errorf("Error getting Result: %v", err)
+		}
+	})
+
+	t.Run("check record", func(t *testing.T) {
+		if recName == "" {
```

Comment:
same as above

Reply:
`recName` is taken from previous tests. We can skip or fail the test here.